README.md

```diff
@@ -79,6 +79,11 @@ aggregation and display of all the components in the Application.
         <td>string</a></td>
         <td>A short, human readable textual description of the Application.</td>
     </tr>
+    <tr>
+        <td>spec.icon</td>
+        <td>string</a></td>
+        <td>A url to the location for an image (e.g., png, gif, or jpeg) representing the Application.</td>
```

Comment:
Can we call out data urls?
https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs

Reply:
+1 on supporting Data URIs. External URLs need to be fetched by the end-user's browser. Fine for cases where the image asset can be hosted publicly, but adds complexity when the app is not intended to be public.

Reply:
+1 for this too

Reply:
How this is supported in Tectonic's system for operators is with the following JSON Schema:

```yaml
icon:
    type: array
    description: Icon which should be rendered with the application information
    required:
    - base64data
    - mediatype
    properties:
    base64data:
        type: string
        description: Base64 binary representation of the icon image
        pattern: ^(?:[A-Za-z0-9+/]{4}){0,16250}(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$
    mediatype:
        type: string
        description: Mediatype for the binary data specified in the base64data property
        enum:
        - image/gif
        - image/jpeg
        - image/png
        - image/svg+xml
```

We actually avoid remote hosted URLs for cross-site request reasons, if you were to display these in a dashboard.

Reply:
@jzelinskie, I'm wondering, why not follow the data URI syntax where media type and encoded data are encoded in a single string?

The advantage of conforming to URI syntax is that dashboards can then choose to implement other mechanisms for fetching images: external URLs, storage buckets, or perhaps other intra-cluster storage mechanisms.

Reply:
We explicitly wanted to limit what could be acceptable input as an icon. This way a dashboard knows it has a finite number of things that it needs to support in order to support all icons. It was us choosing to be slightly more opinionated in order to make building tools that support the resource easier.